### PR TITLE
40K 12 more

### PR DESCRIPTION
**Pull request by Simisays**
Tyranid Harridan's token is added in an earlier PR

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/grey_knight_paragon.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Name:Grey Knight Paragon
+ManaCost:4 W
+Types:Creature Astartes Knight
+PT:4/4
+K:Flash
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDestroy | TriggerDescription$ Rites of Banishment — When CARDNAME enters the battlefield, destroy target attacking creature. If that creature is a Demon, exile it instead.
+SVar:TrigDestroy:DB$ Destroy | ValidTgts$ Creature.attacking | TgtPrompt$ Select target attacking creature | SubAbility$ DBExile | SpellDescription$ Destroy target attacking creature. Draw a card.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/grey_knight_paragon.txt:7]: Also needs Condition to not destroy when exiling

File: forge-gui/res/cardsfolder/upcoming/thunderwolf_cavalry.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Name:Thunderwolf Cavalry
+ManaCost:4 W
+Types:Creature Astartes Warrior
+PT:4/4
+K:First Strike
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | Execute$ TrigPutCounter | CombatDamage$ True | TriggerDescription$ Crushing Teeth — Whenever CARDNAME deals combat damage to a player, put a +1/+1 counter on each other creature you control.
+SVar:TrigPutCounter:DB$ PutCounterAll | ValidCards$ Creature.YouCtrl+Other | CounterType$ P1P1 | CounterNum$ 1
+SVar:HasAttackEffect:TRUE
+DeckHas:Ability$Counter
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/thunderwolf_cavalry.txt:9]: DeckHas:Ability$Counters

File: forge-gui/res/cardsfolder/upcoming/tomb_fortress.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Name:Tomb Fortress
+ManaCost:no cost
+Types:Land
+K:CARDNAME enters the battlefield tapped.
+A:AB$ Mana | Cost$ T | Produced$ B | SpellDescription$ Add {B}.
+A:AB$ Mill | Cost$ 2 B B B T Exile<1/CARDNAME> | Defined$ You | SorcerySpeed$ True | NumCards$ 4 | SubAbility$ DBReturn | SpellDescription$ Mill four cards, then return a creature card from your graveyard to the battlefield. Activate only as a sorcery.
+SVar:DBReturn:DB$ ChangeZone | Origin$ Graveyard | Destination$ Battlefield | ChangeType$ Creature.YouCtrl | ChangeNum$ 1 | Hidden$ True | Tapped$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/tomb_fortress.txt:7]: Doesn't say Tapped

File: forge-gui/res/cardsfolder/upcoming/triarch_praetorian.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Name:Triarch Praetorian
+ManaCost:1 B
+Types:Artifact Creature Necron
+PT:2/1
+K:Flying
+T:Mode$ ChangesZone | Origin$ Graveyard | Destination$ Battlefield | TriggerZones$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDraw | TriggerDescription$ Dynastic Codes — When CARDNAME enters the battlefield from a graveyard, you draw two cards and you lose 2 life.
+SVar:TrigDraw:DB$ Draw |  NumCards$ 2 | SubAbility$ DBLoseLife
+SVar:DBLoseLife:DB$ LoseLife | NumLife$ 2
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/triarch_praetorian.txt:8]: NumLife doesn't exist for this

File: forge-gui/res/cardsfolder/upcoming/triarch_stalker.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Name:Triarch Stalker
+ManaCost:3 B B
+Types:Artifact Creature Necron
+PT:4/5
+S:Mode$ Continuous | Affected$ Creature.attacking ChosenPlayer | AddKeyword$ Menace | Description$ Creatures attacking the last chosen player have menace.
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigChoose | TriggerZones$ Battlefield | TriggerDescriptionLure the Unwary — At the beginning of combat on your turn, choose an opponent.
+SVar:TrigChoose:DB$ ChoosePlayer | Defined$ You | Choices$ Player.Opponent | SetChosenPlayer$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/triarch_stalker.txt:7]: SetChosenPlayer doesn't exist

File: forge-gui/res/cardsfolder/upcoming/trygon_prime.txt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+Name:Trygon Prime
+ManaCost:2 G U
+Types:Creature Tyranid
+PT:4/4
+K:Vigilance
+T:Mode$ Attacks | ValidCard$ Creature.Self | Execute$ TrigCounters | TriggerDescription$ Whenever CARDNAME attacks, put a +1/+1 counter on each attacking creature.
+SVar:TrigCounters:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBCounter
+SVar:DBCounter:DB$ PutCounter | ValidTgts$ Creature.attacking+Other | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBPump
+SVar:DBPump:DB$ Pump | Defined$ ParentTarget | KW$ HIDDEN Unblockable
+DeckHas:Ability$Counter
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/trygon_prime.txt]: Ability$Counter**s**

File: forge-gui/res/cardsfolder/upcoming/trygon_prime.txt [hunk 7]
```diff
@@ -0,0 +1,12 @@
+Name:Trygon Prime
+ManaCost:2 G U
+Types:Creature Tyranid
+PT:4/4
+K:Vigilance
+T:Mode$ Attacks | ValidCard$ Creature.Self | Execute$ TrigCounters | TriggerDescription$ Whenever CARDNAME attacks, put a +1/+1 counter on each attacking creature.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/trygon_prime.txt:6]: TriggerDescription incomplete

File: forge-gui/res/cardsfolder/upcoming/trygon_prime.txt [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+SVar:TrigCounters:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBCounter
+SVar:DBCounter:DB$ PutCounter | ValidTgts$ Creature.attacking+Other | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBPump
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/trygon_prime.txt:8]: "up to" not handled

File: forge-gui/res/cardsfolder/upcoming/vexilus_praetor.txt [hunk 9]
```diff
@@ -0,0 +1,10 @@
+Name:Vexilus Praetor
+ManaCost:3 W
+Types:Creature Custodes Warrior
+PT:3/4
+K:Flash
+K:Vilance
+S:Mode$ Continuous | Affected$ Creature.IsCommander+YouCtrl | AddKeyword$ Protection from everything | Description$ Aegis of the Emperor — Commanders you control have protection from everything.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/vexilus_praetor.txt:7]: Doesn't say creature

File: forge-gui/res/cardsfolder/upcoming/triarch_stalker.txt (hunk 5 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/triarch_stalker.txt:7]: I just copied and changed the script of Beckonin Will O Wisp should I change it there as well ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/triarch_stalker.txt:7]: yea, thanks

File: forge-gui/res/cardsfolder/upcoming/grey_knight_paragon.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/grey_knight_paragon.txt]: This wording probably doesn’t mean replacement effect

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/grey_knight_paragon.txt]: Any pointers to similar cards I should look at ?

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/grey_knight_paragon.txt]: I think you want to use Branch effect – Grisly Sigil might help

If you can't sort it just ping me, I love Branch effect 😁

File: forge-gui/res/cardsfolder/upcoming/thunderwolf_cavalry.txt [hunk 10]
```diff
@@ -0,0 +1,10 @@
+Name:Thunderwolf Cavalry
+ManaCost:4 W
+Types:Creature Astartes Warrior
+PT:4/4
+K:First Strike
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | Execute$ TrigPutCounter | CombatDamage$ True | TriggerDescription$ Crushing Teeth — Whenever CARDNAME deals combat damage to a player, put a +1/+1 counter on each other creature you control.
+SVar:TrigPutCounter:DB$ PutCounterAll | ValidCards$ Creature.YouCtrl+Other | CounterType$ P1P1 | CounterNum$ 1
+SVar:HasAttackEffect:TRUE
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/thunderwolf_cavalry.txt:8]: remove this line, this isn't exactly an attack effect

File: forge-gui/res/cardsfolder/upcoming/thunderwolf_cavalry.txt (hunk 2 above)
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/thunderwolf_cavalry.txt]: Needs to be Counter**s**

File: forge-gui/res/cardsfolder/upcoming/tomb_blade.txt [hunk 11]
```diff
@@ -0,0 +1,12 @@
+Name:Tomb Blade
+ManaCost:4 B B
+Types:Artifact Creature Necron
+PT:5/4
+K:Flying
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | Execute$ TrigLoseLife | CombatDamage$ True | TriggerDescription$ Whenever CARDNAME deals combat damage to a player, that player loses life equal to the number of creatures they control unless they sacrifice a creature.
+SVar:TrigLoseLife:DB$ LoseLife | Defined$ TriggeredTarget | UnlessCost$ Sac<1/Creature> | UnlessPayer$ TriggeredTarget | LifeAmount$ X
+SVar:X:TriggeredTarget$Valid Creature
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/tomb_blade.txt:8]: seems to be missing `YouCtrl`

File: forge-gui/res/cardsfolder/upcoming/tomb_fortress.txt (hunk 3 above)
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/tomb_fortress.txt]: `YouOwn` is better here
Remove `Tapped$`

File: forge-gui/res/cardsfolder/upcoming/grey_knight_paragon.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/grey_knight_paragon.txt]: Thanks with the brach effect it's working

**Review by paulsnoops — APPROVED** [review]

**Pull request merged**